fle-internal/kolibri-instant-schools-plugin
# Add "reset password via SMS" workflow

**Providing Phone Number to Get SMS**
![get token](https://user-images.githubusercontent.com/10248067/31907495-d94a168c-b7e8-11e7-8e87-91f93ba1856c.gif)

**Resetting password**
![reset pw](https://user-images.githubusercontent.com/10248067/31907502-ddb2e15e-b7e8-11e7-9ff3-5d0f92b11e15.gif)


To test:

1. Install the SMS simulator (see the README in the `/smpp` folder of this repo)
1. Go to http://localhost:8000/user/api/phonesignup/ and create an account with your phone number as `username`. IMPORTANT! Save the actual username (e.g. `9ff4e87a0673406191ffe8b0a37777` that will appear as soon as the account is created.)
1. Go to http://localhost:8000/user/#/signin
1. Click "Reset password"
1. Happy-path: (Hopefully) go through the dialogue without any trouble
1. Non-happy paths: a) enter a bogus phone number, b) enter a legit phone number, but with SMPP simulator turned off. Should get some explanation from UI why reset attempt didn't work.
1. If finished happy path, go to terminal window with SMPP and look for a message like this: `please click the following link: http://localhost:8000/user/#/passwordreset/8586035765/374992445640`
1. Follow that link
1. Happy-path: Reset your password using the dialogue, then try logging-in again, but using the hash username `9ff4e87a0673406191ffe8b0a37777`, not your phone number! (phone isn't being swapped out for real username behind the scenes)
1. Non-happy path: a) Go back to the link after resetting password, b) change some of the parameters (e.g. phone number and token) in the URL. The UI will give you a generic "Token Expired" message (same message even if there was no token to begin with).

Other details:
1. Validations for text fields
1. Throttle network tab and make sure forms disable when needed
